reference/reference_test.go

```diff
@@ -169,7 +184,7 @@ func TestReferenceParse(t *testing.T) {
 			failf("unexpected parse error: %v", err)
 			continue
 		}
-		if repo.String() != testcase.input {
+		if !strings.EqualFold(repo.String(), testcase.input) {
```

Comment:
Not a fan of this, `String()` is intended to return the original input. This makes reference not safe to be used a map key.